core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPathfinding.kt

```diff
@@ -204,9 +208,10 @@ class STDCMPathfinding(
                 step.locations
                     .filter { it.edge == edge.block }
                     .mapNotNull { edge.edgeOffsetFromBlock(it.offset) }
-            assert(locationOnEdge.size <= 1)
-            if (locationOnEdge.isNotEmpty()) {
-                res.add(EdgeLocation(edge, locationOnEdge.first()))
+                    .minOrNull()
+            // Sometimes a step has several locations on the same edge, we just pick the first
+            if (locationOnEdge != null) {
```

Comment:
Do we want the first in the list or the first ordered by block offset ?

Reply:
I first wrote `.sorted().firstOrNull()`, but my IDE suggested `.minOrNull()`, that's what it does

Reply:
Actually legit did not see the minOrNull, guess i'm just blind :D